### PR TITLE
don't break when file type was detected but no extension was provided

### PR DESCRIPTION
**Pull request by nanu-c**

### CODE REVIEW

File: app/store/attachment.go [hunk 1]
```diff
@@ -40,12 +40,15 @@ func SaveAttachment(a *textsecure.Attachment) (Attachment, error) {
 		extension, err := mime.ExtensionsByType(a.MimeType)
 		if err != nil {
 			log.Debugln("[axolotl] could not detect file extension", a.MimeType)
-			if extension == nil {
+			if extension == nil || len(extension) == 0 {
 				extension = []string{""}
```
Comment (Flaburgan) [inline app/store/attachment.go:44]: Do you use that `extension` variable for anything else later? All that code seems complicated to me, I would have written something like
```go
if fileName == "" {
  fileName = helpers.RandomString(10)
  extension, err := mime.ExtensionsByType(a.MimeType)
  if err != nil {
    log.Debugln("[axolotl] could not detect file extension", a.MimeType)
  } else {
    fileName +=  extension[0]
  }
}
```
We need unit tests...

**Review by Flaburgan — CHANGES_REQUESTED** [review]

File: app/store/attachment.go (hunk 1 above)
Comment (nanu-c) [reply app/store/attachment.go:44]: There is then missing the case, when `mime.ExtensionsByType()` doesn't throw an error but also doesn't provides an extension, that's the case in voice messages.

Comment (nanu-c) [reply app/store/attachment.go:44]: I agree with the unit tests

Comment (Flaburgan) [reply app/store/attachment.go:44]: >There is then missing the case, when mime.ExtensionsByType() doesn't throw an error but also doesn't provides an extension, that's the case in voice messages.

Well, you're current code isn't doing anything more about that case, as it puts `""` in extension and then concats it at the end of `fileName` which looks useless to me. Or am I missing something?

Comment (Flaburgan) [reply app/store/attachment.go:44]: Oh yeah I'm missing the if sorry, so the code should be 
```go
if fileName == "" {
  fileName = helpers.RandomString(10)
  extension, err := mime.ExtensionsByType(a.MimeType)
  if err != nil {
    log.Debugln("[axolotl] could not detect file extension", a.MimeType)
  } else if len(extension) > 0 {
    fileName +=  extension[0]
  }
}
```

**Comment by nanu-c** [comment]
I changed it now and you have written your first lines of go code :)

**Review by Flaburgan — APPROVED** [review]

**Comment by Flaburgan** [comment]
Aaaaaaand it works! Approving this!

**Pull request merged**